### ISSUE TITLE
Invitation to stakeholders; comments, stories, requirements, feature requests, etc.

### ISSUE THREAD

**Issue by LithiumToast** [Styxz/Assetyard#6]
Hello. 

For this open source project I'm going to be doing something different. I'm going to be applying knowledge from a software architecture class I toke at university. We used this book in the class: [Software Architecture in Practice, 3rd Edition](https://www.amazon.com/Software-Architecture-Practice-3rd-Engineering/dp/0321815734).

I'm opening the floor to comments, stories, requirements, etc, from stakeholders for a Content Pipeline. A stakeholder for this project is anyone who uses, is affected by, or programs the system or application. This would include game developers who directly use the Content Pipeline as a tool; the end users or people who use the resulting game or simulation the game developers produce for various reasons; open source developers who help contribute and develop the system or application; etc.

From this information I will do the following:
- Derive business goals
- Derive architecturally significant requirements (ASRs)
- Derive the non-functional requirements or better called "quality attributes"
- Create a utility tree that captures the ASRs and quality attributes and prioritize them against the business goals
- Create quality attribute scenarios

## Background information from the book (Skip this section if you want)
"An architecturally significant requirement (ASR) is a requirement that will have a profound effect on the architecture - that is, **the architecture might well be dramatically different in the absence of such a requirement**. **You cannot hope to design a successful architecture if you do not know the ASRs.**"

"... as we observe architects going about their duties, we notice that the **first thing they do is start talking to the important stakeholders**. They're gathering the information they need to produce the architecture that will respond the project's needs - wether or not this information has already been identified."

"Many projects don't create or maintain the kind of requirements document that professors in software engineering classes or authors of traditional software engineering books love to prescribe. Whether requirements are specified using the 'MoSCoW' style (must, should, could, won't), or as a collection of 'user stories', neither of these is much help in nailing down quality attributes. Furthermore, no architect just sits and waits until the requirements are 'finished' before starting to work. The architect **must begin while the requirements are still in flux**."

"Requirement documents often fail an architect in two ways:
 - First, most of what is in a requirements specification does not affect the architecture. Architectures are mostly driven or 'shaped' by quality attribute requirements. These determine and contain the most important architectural decisions. And yet the vast bulk of most requirements specifications is focused on the required features and functionality of a system, which shape the architecture the least. The best software engineering practices do prescribe capturing quality attribute requirements.
In practice, though, we rarely see adequate capture of quality attribute requirements. How many times have you seen a requirement of the form 'The system shall be modular' or 'The system shall exhibit high usability' or 'The system shall meet users' performance expectations'? **These are not requirements, but in the best case they are invitations for the architect o begin a conversation about what the requirements in these areas really are**.
- Second, much of what is useful to an architect is not in even the best requirements document. Many concerns that drive an architecture do not manifest themselves at all as observables in the system being specified, and so are not the subject of requirements specifications. **ASRs often derive from the business goals in the development organization itself.**

"**Stakeholders often have no idea what quality attributes they want in a system, and no amount of nagging is going to suddenly install the necessary insight**. ... **Interviewing the relevant stakeholders is the surest way to learn what they know and need.** Once again it behooves a project to capture this critical information in a systematic, clear, and repeatable way."

"The results ... can be used to do the following:
- Refine system and software requirements
- Understand and clarify the system's architectural drivers
- Provide rationale for why the architect subsequently made certain design decisions
- Guide the development of prototypes
- Influence the order in which the architecture is developed"

## Template for stakeholder responses
- Introduce yourself; sate your background and what is your relationship to the application or system being built; what kind of stakeholder are you?
- Define your business needs; what are your broad functional requirements, constraints, and known non-functional or quality attribute requirements for the system or application?
- Express a scenario or story representing your concerns with respect to the system or application; how do you envision using the system or application?; Be sure to include a stimulus and a response. A stimulus is an event or action that invokes the system or application to have a response.

## Results
I will merge stakeholder responses that are similar and post them along with derived results in a separate issue periodically for iterations of architecture development. Stakeholders can vote (through emojis) and discuss the results if they so wish so I can gauge if the results are good reflection of their input.
